apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-previewing-theme.ts

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react';
+import wpcom from 'calypso/lib/wp';
+import { currentlyPreviewingTheme, PREMIUM_THEME, WOOCOMMERCE_THEME } from '../utils';
+import type { Theme } from 'calypso/types';
+
+/**
+ * Get the theme type.
+ * This only support WooCommerce and Premium themes.
+ */
+const getThemeType = ( theme?: Theme ) => {
+	const theme_software_set = theme?.taxonomies?.theme_software_set;
+	const isWooCommerceTheme = theme_software_set && theme_software_set.length > 0;
```

Comment:
I think a theme is a WooCommerce theme if it has `woo-on-plans` in the software sets, can you double check?

Reply:
You seem to be right (I might look into deeper a bit). 
The `getThemeType` selector https://github.com/Automattic/wp-calypso/blob/8ad064cf187f233781a992e63b5e5944c799bad1/client/state/themes/selectors/get-theme-type.js#L27-L29 is using the same logic, so It'd be nice to fix it as well.

Reply:
The `getThemeType` was first introduced in https://github.com/Automattic/wp-calypso/pull/75396 (and then extracted as a selector in https://github.com/Automattic/wp-calypso/pull/77550/). 
I believe we are not purposely using `doesThemeBundleSoftwareSet` for the `WOOCOMMERCE_THEME` condition. 🙂  I'll address this in another PR so that it checks `woo-on-plans` (e.g., https://github.com/Automattic/wp-calypso/blob/33afc9ac13d7a7b729008d82b3182dce4a3dab03/client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx#L442-L444, https://github.com/Automattic/wp-calypso/blob/33afc9ac13d7a7b729008d82b3182dce4a3dab03/packages/data-stores/src/starter-designs-queries/use-starter-designs-query.ts#L98-L100). @fushar @Automattic/t-rex

Reply:
`getThemeType` is being refactored in a different way in https://github.com/Automattic/wp-calypso/pull/84483/files#diff-136055987b5718b1aa6eb57c3cda1dad677dd540def25ff13f9985936cb14019 I haven't dug into it yet but it looks like it is using the concept of "bundled theme" rather than specific woo/sensei theme types